### PR TITLE
MueLu: kokkos version of SaP satisfy constraints

### PR DESCRIPTION
**Pull request by rstumin**
Auto-PR for SHA 705afec

### CODE REVIEW

**Review by jhux2 — APPROVED** [review]
lgtm

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp [hunk 1]
```diff
@@ -48,6 +48,11 @@
 
 #ifdef HAVE_MUELU_KOKKOS_REFACTOR
 
+#ifdef out
+#include "KokkosKernels_Handle.hpp"
+#include "KokkosSparse_spgemm.hpp"
+#include "KokkosSparse_spmv.hpp"
+#endif
```
Comment (jhux2) [inline packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp:55]: What's this stuff?

**Review by jhux2 — APPROVED** [review]
lgtm

**Review by jhux2 — DISMISSED** [review]

**Review by jhux2 — DISMISSED** [review]
My name is jhux2, and I approve this PR.

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp [hunk 2]
```diff
@@ -271,53 +271,56 @@ namespace MueLu {
...
-   Rsum = Kokkos::View<Scalar*, Device>("Rsum",nPDEs);
-   nPositive = Kokkos::View<size_t*, Device>("nPositive",nPDEs);
+   Kokkos::View<SC**, Device> ConstraintViolationSum;
+   Kokkos::View<SC**, Device> Rsum;
+   Kokkos::View<size_t**, Device> nPositive;
+
+  constraintKernel(LO nPDEs_,local_matrix_type localP_) : nPDEs(nPDEs_), localP(localP_)
+   {
+     ConstraintViolationSum = Kokkos::View<SC**, Device>("ConstraintViolationSum", localP_.numRows(), nPDEs);
+     Rsum = Kokkos::View<SC**, Device>("Rsum", localP_.numRows(), nPDEs);
```
Comment (jhux2) [inline packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp:291]: Is this the ugly fix?

Comment (lucbv) [reply packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp:291]: Yep, I have to ask someone else on the kokkos team about thread local memory when one is not using a team policy...

**Review by jhux2 — APPROVED** [review]
Seems fine.

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp (hunk 2 above)
Comment (brian-kelley) [reply packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_kokkos_def.hpp:291]: I don't think there's any way to use shared mem in a RangePolicy. Are you worried about lost performance from having ``ConstraintViolationSum`` and ``Rsum`` be in global memory instead? I would not expect it to make that much difference. You could also add ``Kokkos::LayoutRight`` to make the rows contiguous in memory, then caching will be better.

**Pull request merged**